### PR TITLE
Better dice rolls

### PR DESCRIPTION
**Pull request by tawilkinson**
**Describe this PR and why it should be merged:**

Currently Steve does not support a few useful features for playing D&D games (such as our ongoing modmin game). These are:

- Static modifiers to represent ability checks/extra damage.
- Summation of all rolls when rolling multiple dice.\*

Also I made improvements to how roll results are displayed. [Role players are a superstitious bunch](https://www.darthsanddroids.net/episodes/0099.html) so the output is now more verbose:

- When using either version of the keep functionality a list of ~struck through~ rolls is prepended so you can see the rolls that were ignored.
- When using exploding dice the list of all rolls and the sum is shown (alongside the 💥 emoji) inside [square brackets].
- When rolling multiple different sets of dice (using the `|` operator) the command is shown within backticks to distract less from the result.
- All results have been moved outside backticks to allow for nicer formatting:
    - The sum of the dice is shown in **bold**
    - Exploded dice are emphasised in **bold** (but within the square brackets mentioned above).
    - The summed rolls for multiple dice are displayed _italicised_ and (in brackets).
    - Static modifiers are displayed in plain text to make them clearly distinct from rolled values.

While I was there I added 3 lines to the Readme that may be helpful to new developers (thanks for the help setting things up Jonathan!).

Example output in Discord:
all the tests

\* It may be possible that someone does not want to know the sum of their dice when rolling a bunch of dice. We show all rolls explicitly as well as a sum so the old functionality is not lost.

**Status**
- [x] This PR has been tested and complies with this project's ESLint rules
- [x] This PR adds/modifies a command
	- [x] The `examples` and `helpUsage` options have been created/updated accordingly

**Semantic Versioning**
- [ ] This PR features bug fixes, or only style changes/refactorings, or no code changes
- [x] This PR adds features for users
- [ ] This PR switches this project to a new a new Discord API library/new Discord.js framework

### CODE REVIEW

File: src/commands/Fun/Random/roll.ts [hunk 1]
```diff
@@ -27,10 +50,11 @@ export default class extends SteveCommand {
...
-			examples: ['roll 1d6', 'roll d20', 'roll 5d10!', 'roll 1d8|4d6', 'roll 6d12k1', 'roll 6d12kl2'],
-			extendedHelp: oneLine`Use standard dice notation. You can roll up to 10 dice with up to 1,000 sides each.
+			examples: ['roll 1d6', 'roll d20', 'roll 4d6-5', 'roll 5d10!', 'roll 1d8|4d6', 'roll 6d12k1', 'roll 6d12kl2'],
+			extendedHelp: oneLine`Using standard dice notation: You can roll up to 10 dice with up to 1,000 sides each.
 				Add a \`!\` at the end of your roll to use exploding dice.
 				To keep the highest n, add \`k<n>\`; to keep the lowest n, add \`kl<n>\` (with n < amount of dice).
+				You can add a static positive or negative modifier to the sum of the roll using \`+<n>\` or \`-<n>\`.
 				You can do multiple rolls at once, separated by \`|\`.`,
 			usage: '<spec:dice> [...]',
 			helpUsage: '<number of dice>d<number of sides>'
```
Comment (jwford) [inline src/commands/Fun/Random/roll.ts]: okay due to the big update recently, this is where the bulk of the necessary changes are gonna be. `helpUsage` and `examples` are unecessary now, as all of the command helps are in the `en-US` language file and everything has been internationalized. so i'd just port your additions to `examples` and `extendedHelp` over to that file.

File: src/commands/Fun/Random/roll.ts [hunk 2]
```diff
@@ -92,13 +223,16 @@ export default class extends SteveCommand {
 		if (results.length === 1) {
 			const result = results[0];
 			const emoji = getEmoji(result.spec);
-			const message = `${emoji} You rolled: \`${result.rolls.join(', ')}\` ${emoji}`;
+			// just display the message when a single roll is made
+			const message = `${emoji} You rolled: ${result.message} ${emoji}`;
```
Comment (jwford) [inline src/commands/Fun/Random/roll.ts]: needs to be internationalized as well

**Review by jwford — CHANGES_REQUESTED** [review]

File: src/commands/Fun/Random/roll.ts (hunk 1 above)
Comment (tawilkinson) [reply src/commands/Fun/Random/roll.ts]: Fixed and passing checks in 2a9d846be6c76b21007c47ebb0851428269a7c3a

File: src/commands/Fun/Random/roll.ts (hunk 2 above)
Comment (tawilkinson) [reply src/commands/Fun/Random/roll.ts]: A few lines below needed updating too. Should be fixed  with checks passing in ae7df243c747f41e68ae580918f4f94dffbf01a1

**Comment by tawilkinson** [comment]
I believe the internationalisation changes have been made but it needs testing and my test PC is offline and it is almost midnight. Will test tomorrow.

File: src/commands/Fun/Random/roll.ts (hunk 2 above)
Comment (tawilkinson) [reply src/commands/Fun/Random/roll.ts]: Tested and working

File: src/commands/Fun/Random/roll.ts (hunk 1 above)
Comment (tawilkinson) [reply src/commands/Fun/Random/roll.ts]: Tested!

**Comment by tawilkinson** [comment]
Tested on my test server. I believe this is ready to merge

**Review by jwford — APPROVED** [review]

**Pull request merged**